packages/jbrowse-plugin-apollo/src/FeatureDetailsWidget/TranscriptWidgetEditLocation.tsx

```diff
@@ -159,31 +166,146 @@ export const TranscriptWidgetEditLocation = observer(
           continue
         }
         if (isMin && oldLocation === child.min) {
-          const change = new LocationStartChange({
-            typeName: 'LocationStartChange',
-            changedIds: [child._id],
-            featureId: feature._id,
-            oldStart: child.min,
-            newStart: newLocation,
-            assembly,
-          })
-          changeManager.submit(change).catch(() => {
-            notify('Error updating feature start position', 'error')
-          })
+          newLocation = newLocation - 1
+          if (newLocation < 0) {
+            notify('Start location should be greater than 0', 'error')
+            return
+          }
+          if (index > 0) {
+            for (let i = index - 1; i >= 0; i--) {
+              const prevLoc = transcriptExonParts[i]
+              if (prevLoc.type === 'exon') {
+                if (prevLoc.max > newLocation) {
+                  notify(
+                    'Start location should be greater than previous exon end location',
+                    'error',
+                  )
+                  return
+                }
+                break
+              }
+            }
+          }
+
+          const changes = []
+          if (index === 0) {
+            const transcriptStart = feature.min
+            const geneStart = feature.parent?.min
+            if (newLocation < transcriptStart) {
+              if (geneStart && newLocation < geneStart) {
+                // gene start
+                changes.push(
+                  new LocationStartChange({
+                    typeName: 'LocationStartChange',
+                    changedIds: [feature.parent._id],
+                    featureId: feature.parent._id,
+                    oldStart: feature.parent.min,
+                    newStart: newLocation,
+                    assembly,
+                  }),
+                )
+              }
+
+              // transcript start
+              changes.push(
+                new LocationStartChange({
+                  typeName: 'LocationStartChange',
+                  changedIds: [feature._id],
+                  featureId: feature._id,
+                  oldStart: feature.min,
+                  newStart: newLocation,
+                  assembly,
+                }),
+              )
+            }
+          }
+
+          // exon start
+          changes.push(
+            new LocationStartChange({
+              typeName: 'LocationStartChange',
+              changedIds: [child._id],
+              featureId: feature._id,
+              oldStart: child.min,
+              newStart: newLocation,
+              assembly,
+            }),
+          )
+
+          // Submit all changes sequentially. gene -> transcript -> exon
+          for (const change of changes) {
```

Comment:
These should all be in a `changes` attribute of a single `LocationStartChange` instead of three different `LocationStartChange`s. So it would look something like this:

```js
new LocationStartChange({
  typeName: 'LocationStartChange',
  changedIds: ['feature1Id', 'feature2Id'],
  assembly,
  changes: [
    { featureId: 'feature1Id', oldStart: 500, newStart: 490 },
    { featureId: 'feature2Id', oldStart: 500, newStart: 490 },
  ],
})
```

Same for the `LocationEndChange`s.